libs/wingsdk/src/cloud/table.ts

```diff
@@ -0,0 +1,134 @@
+import { Construct } from "constructs";
+import { fqnForType } from "../constants";
+import { App, Resource } from "../core";
+import { Json } from "../std";
+
+/**
+ * Global identifier for `Table`.
+ */
+export const TABLE_FQN = fqnForType("cloud.Table");
+
+/**
+ * Table column types
+ */
+export enum ColumnType {
+  /** string type */
+  STRING,
+  /** number type */
+  NUMBER,
+  /** bool type */
+  BOOLEAN,
+  /** date type */
+  DATE,
+  /** json type */
+  JSON,
+}
+
+/**
+ * Properties for `Table`.
+ */
+export interface TableProps {
+  /**
+   * The table's columns.
+   * @default undefined
+   */
+  readonly columns: { [key: string]: ColumnType };
+  /**
+   * The table's primary key. No two rows can have the same value for the
+   * primary key.
+   * @default undefined
+   */
+  readonly primaryKey: string;
+}
+
+/**
+ * Represents a NoSQL database table that can be used to store and query data.
+ * @inflight `@winglang/sdk.cloud.ITableClient`
+ */
+export abstract class Table extends Resource {
+  /**
+   * Create a new `Table` instance.
+   * @internal
+   */
+  public static _newTable(
+    scope: Construct,
+    id: string,
+    props: TableProps
+  ): Table {
+    return App.of(scope).newAbstract(TABLE_FQN, scope, id, props);
+  }
+
+  /**
+   * Table primary key name
+   */
+  public readonly primaryKey: string;
+  /**
+   * Table columns
+   */
+  public readonly columns: { [key: string]: ColumnType };
```

Comment:
I tryed this before but gives an JSii error in `TableProps`, so I changed to this object `{ [key: string]: ColumnType }`

`src/cloud/table.ts:35:12 - error JSII3001: Type "Map" cannot be used as the property type because it is private or @internal`

Reply:
Yeah -- unfortunately JSII doesn't allow public APIs using a many built-in JavaScript types like `Set`, `Map`, `Date`, `Error`, etc. so I think a plain object map will have to do. 😕 (I think the reason is that many of these types don't have equivalents in the other JSII target languages)

Reply:
Aw thanks JSii! lol yea @marciocadev feel free to disregard all my blabbering about Map types 😁

Reply:
Don't worry man

I made the same mistake when I wrote these classes


JSii hurt my feelings too 😂😂😂